### DISCUSSION TITLE
Questions on Soprano sax mouthpiece, blowing sensitivity, PC-speaker connectivity

### DISCUSSION THREAD

**Discussion by nh2** [cardonabits/haxo-hw#45] [Q&A] [answered]
My CrowdSupply Haxophone just arrived, so now I'll bombard you with some questions :D

* Would it be difficult to make a mouthpiece adapter that works with a Soprano sax mouthpiece (which is the only one I currently have)? I have a 3D printer ready :)
* Is it possible to adjust the blowing sensitivity? Currrently without sax mouthpiece, I have to blow in quite hard.
* When connecting to a PC via the 3.5mm jack (e.g. to use a loudspeaker system), are there some tipps on whether to use a Mic or Line-In connector? I read some of your posts on headphone impedance, but I'm not sure how to get the necessary info for random desktop/laptop sound cards. Maybe I should better play via USB-Midi instead? If yes do you have some tips to "forward" the Midi to a Linux PC so that it plays the same way as it does on the Haxophone's Linux? In your video you show it with Mac software.

Thanks!

**Comment by jcard0na** [comment]
Hi @nh2 

Yeah, bring them on!

> Would it be difficult to make a mouthpiece adapter that works with a Soprano sax mouthpiece (which is the only one I currently have)? I have a 3D printer ready :)

If you are familiar with FreeCad, very easy.  Download [this file](https://github.com/cardonabits/haxo-hw/blob/main/models/neck.FCStd) and change the neck diameter in Sketch004 from 16mm to your desired dimensions.

![Screenshot 2023-12-20 at 14 44 27](https://github.com/cardonabits/haxo-hw/assets/676181/f46bdf19-d651-4ede-afba-00d8458c4008)

Then print away!

> Is it possible to adjust the blowing sensitivity? Currently without sax mouthpiece, I have to blow in quite hard.

Hmmm, is that should not be the case.  You mean that the volume you hear on your speaker or headphone is too low unless you blow very hard?  That might be a speaker issue rather than the sensor.  In any case, this is where the pressure sensor code lives:  https://github.com/cardonabits/haxo-rs/blob/main/src/pressure.rs. If you are comfortable with embedded code, you could run that code and see the sensor sensitivity and response.  

> When connecting to a PC via the 3.5mm jack (e.g. to use a loudspeaker system), are there some tips on whether to use a Mic or Line-In connector?  I read some of your posts on headphone impedance, but I'm not sure how to get the necessary info for random desktop/laptop sound cards. 

On a typical sound card, you want to use line-input as it will have lower input impedance than mic-input, and therefore more signal power will be transferred.  On very high-impedance inputs ("mic", but also sometimes labelled "high-Z" or "instrument input" you might have a very high voltage which results in clipped or distorted audio.  However, I don't think you can damage your equipment if you plug the haxophone on the wrong port:  just unplug it right away if the sound quality is bad.

> Maybe I should better play via USB-Midi instead? 

No, you will get better latency out of the analog output.  I only use USB-Midi when I need to capture/transcribe what I play.

> If yes do you have some tips to "forward" the Midi to a Linux PC so that it plays the same way as it does on the Haxophone's Linux? In your video you show it with Mac software.

I do not use any MIDI software on Linux, but I've confirmed that the MIDI device is picked up when the haxophone is connected to a Linux host.  I tested that with `aplaymidi -l`. So whatever MIDI software you want to use should be able to "see" the haxophone as a midi device.

**Reply by nh2** [reply]
Hi Javier, thanks for the quick answer!

> If you are familiar with FreeCad, very easy. Download [this file](https://github.com/cardonabits/haxo-hw/blob/main/models/neck.FCStd) and change the neck diameter in Sketch004 from 16mm to your desired dimensions.

Your graphic shows the current adapter being 16 mm. The Soprano sax mouthpice (I have one Selmer and one Yamaha) hole is 14 mm. This means I'd have to reduce the adapter from 16 mm to perhaps 13 mm. At that point it would touch the 2 smaller holes through which the air goes. So I'd likely have to shrink those.

Can the air holes be shrunk without it affecting the sound, or might other software/hardware adjustments be needed to do that?

> You mean that the volume you hear on your speaker or headphone is too low unless you blow very hard?

I have inspected it a bit more now with a few headphones, and it seems you are right.
When I'm in a low-noise room, I can hear that audio is playing also at "low pressure", and that I can explore the volume range as I blow harder.

So I guess my real issue really is volume control:

In a loud environment (e.g. public transport), I need to blow quite hard to hear myself on good volume with normal headphones (e.g. simple 10 EUR earbuds, or my SteelSeries Arctis 7 in passive cabled mode). It would be nice to be able to "play louder" while still blowing with minimal effort in such cases.

Can I tell the Haxophone to adjust the volume?

Btw, I think the Haxophone still needs some proper "manual". E.g. the keyboard combination to switch instruments is in [`instructions_safety.md`](https://github.com/cardonabits/haxo-hw/blob/main/docs/instructions_safety.md) which isn't quite where I would have initially searched for it (I found it with Github Search then).

> you will get better latency out of the analog output

How big is the latency approximately?

I was thinking to try and hook up the Haxophone to e.g. https://www.reaper.fm (proprietary but cross-platform DAW) to get the full power of a PC for sound generation.

**Reply by jcard0na** [reply, marked as answer]
Hi @nh2,

> Can the air holes be shrunk without it affecting the sound, or might other software/hardware adjustments be needed to do that?

Each of the two holes has a function. The first one transfer the pressure to the sensor, and the other one evacuates (vents) the air.  If you narrow down the holes, you will have to adjust the outer diameter of the silicon tubes that they connect to.  You can experiment with that, the only effect will be on "air resistance" to your blowing, but no effect on sound.

Since the inner diameter of the silicon tubes is 2mm, you could "taper" neck and the inner holes from the current 6mm diameter to a lower value and have no impact on air flow.  That said, doing that is beyond my FreeCad skill level.  But maybe not yours!

Alternatively, you could switch the silicon tubing from the current 6mm OD/2mm ID for a 4mm OD/2mm ID.  This is a pretty standard size.  If you do that, you could reduce the diameter of the holes from 6 to 4mm, the distance between the two from 8 to 6mm and the outer diameter from 16 to 13mm.

> Can I tell the Haxophone to adjust the volume?

No, not really.  You can adjust the gain of the amp on a semi-permanent basis, either to double it or halve it.  See https://github.com/cardonabits/haxo-hw/discussions/43#discussioncomment-7907410

As for volume control proper, the pressure sensor is all there is.  If people request it, I might take a stab implementing this via control messages, like what we do for switching instruments.

> Btw, I think the Haxophone still needs some proper "manual".

Thanks for the feedback.  I just filed #46 ;)

> How big is the latency approximately?

On built-in audio is about 12ms, which is pretty good.  Once you introduce midi, it all depends on your setup, but it will be much longer.  It's good for transcribing/recording music, where you can compensate for latency in software.  But not for live performance.  For that, you want the built-in output.

**Comment by nh2** [comment]
Some more questions:

## "Slur-clicking"

When playing loud, and pressing a key, one can hear a bit of a "clicking" sound. For example, when playing low E loudly, and toggling down to low D a couple times (without stopping the air flow, thus playing a [slur](https://en.wikipedia.org/wiki/Slur_(music))). On a normal saxophone, this would make the note switch more softly, so I wonder if once can reduce this slur-clicking, and whether it is a hardware or software thing.

## Upper side keys distance

The first hardware hacking I might be looking into is for the upper side keys. They feel a bit "far out", e.g. with my medium-sized hands I have to completely let to of all left-hand fingers to press a side key (e.g. the high side-D); I also have to let go of the left-thumb pad.

On the Haxophone the side keys are around 8 cm from the B key.

On e.g. the Soprano, they are much closer, around 4 cm:

![image](https://github.com/cardonabits/haxo-hw/assets/399535/e47d5d30-5973-474d-ac0d-908d05ecf326)

So there I would not have to let go of the whole left hand to press e.g. the middle button.

(This whole topic might be easier to communicate in a video.)

So I am thinking that a good first hardware hack for me could be to try to move the upper side keys closer.

Do you have a tip on how to do that the easiest? Some ideas:

* Use flatter switches and key caps for the side buttons.
* Shift the 90-degree side PCB "further inwards" by replacing the default H12/H13 angle connectors by some (3d-printed ones) that allow it to be further in.
  * This would require replacing the electric connector J8 by some flexible cables.

## Cleaning instructions video

I noticed that some spit comes out of the tube connected to the air sensor. This is likely expected. In the instructions you recommend to clean the tubes. However, I am a bit scared of decoupling the tube from the sensor, as I'd be pulling on the sensor which is held to the PCB only by its soldered feet and perhaps some glue on its lower body.

Could you make a short video to show how you do the cleaning process (also showing how to release and reassemble the air tubes) to give a bit more confidence?

**Reply by jcard0na** [reply]
> When playing loud, and pressing a key, one can hear a bit of a "clicking" sound.

Yes, I am aware that you cannot perfectly slur notes.  That is just a midi limitation, which is used internally to synthesize sound.  However, I found that with good breath control, you can approximate slur playing quite nicely.  For instance, see Nigel's playing here: https://youtu.be/quGswQKbFMA?t=422 (he's a better player than I will ever be, and I cannot sound so smooth, but sharing here for inspiration)

> Do you have a tip on how to [ move the upper side keys closer]? Some ideas

Now that you have a haxophone in your hands you might have realized that my engineering skills are strong in software, adequate in electronics, and pretty basic in mechanical engineering.  Both options you suggest sound reasonable to me.  The good thing of me having poor mechanical chops is that I made the mechanical design **very** simple.  Those L-brackets are as crude as they get, so they are easy to replace, modify or bend to your needs.  

>. Could you make a short video to show how you do the cleaning process (also showing how to release and reassemble the air tubes) to give a bit more confidence?

Sure, I'll queue that up.  But just to give you some confidence, the silicon tube is just pressure fitted onto the sensor.  So pressing down with one finger on the sensor, and slowly pulling on the tube should release it and avoid any strain to the sensor.    There is no glue that you have to worry about.  The only thing you have to ensure is that you pull perpendicularly to the sensor port to avoid breaking that.  (You will not, but if you did break it, you it's not hard to replace but it will set you back approx $18 to buy an new sensor) 

![Screenshot 2023-12-20 at 18 43 17](https://github.com/cardonabits/haxo-hw/assets/676181/f981bf86-b790-42ab-bd2c-0fa37300f88c)

The same applies to the other end that is inserted into the neck.  Just pull.

**Comment by thuntjazz** [comment]
> * Is it possible to adjust the blowing sensitivity? Currrently without sax mouthpiece, I have to blow in quite hard.

I put a cheap alto mouthpiece and a worn out reed (it doesn't have to vibrate, it's just for the right feel) on my neck, and I like the breath control much better.  A thought I had, but haven't yet experimented with:  you could, theoretically, use the reed position on the mouthpiece as a bit of a resistance adjustment.  The reed's normal position, with the tip even with the tip of the mouthpiece, would give you the most resistance, while moving the reed down and exposing more of the chamber of the mouthpiece would allow you to push more air.  Of course, you're still ultimately limited by the size of the holes in the neck, and I haven't tested this yet to see if reed position makes any difference at all.  But I will try it soon and report back  :-)

**Reply by nh2** [reply]
This is a nice idea.

If the quality of the mouthpiece does not matter so much for a Haxophone as it matters for a real saxophone, I might first try to simply 3D-print an Alto or Tenor saxophone mouthpiece, e.g. this: https://www.thingiverse.com/thing:14495

This might be simpler than adjusting the parts on the Haxophone.

**Reply by thuntjazz** [reply]
Yeah, the mouthpiece and reed do nothing as far as tone quality is concerned, so it doesn't matter how good they are.  It's just about, at least for me, making it feel more like a saxophone.  The resistance/backpressure feels remarkably similar to my alto now, even though the reed isn't vibrating.

I also toyed with the idea of using a recorder-style mouthpiece without a reed at all, but the one I had lying around doesn't fit.  I can definitely see a mouthpiece that doesn't need a reed being even more preferable, particularly for non-saxophonists.  That kind of piece would, perhaps, be even easier to dial in the amount of resistance felt by the player.

3D printed mouthpieces have been around for a while now, and more and more people have started playing them.  The SYOS company in France is probably the most well known manufacturer of 3D printed mouthpieces, including custom-printed signature models for some great players.  There is a bit of chatter out there regarding placing ABS plastics in your mouth and health concerns, but there's little research out there to confirm or refute any of it...it may, honestly, be some negative marketing by the more traditional mouthpiece makers  :-)

**Reply by Utopiah** [reply]
Apparently this model https://github.com/cardonabits/haxo-hw/discussions/65#discussioncomment-8042691 is a good fit, will report back in few hours after printing and testing.

**Reply by Utopiah** [reply]
![image](https://github.com/cardonabits/haxo-hw/assets/1076655/2dcc1d07-5007-4d06-8c90-e63111593ddb)

**Reply by jcard0na** [reply]
That hint of fluo washi tape is very striking between all that blackness!

**Comment by Ekolide** [comment]
![haxo-w-mouthpiece](https://github.com/cardonabits/haxo-hw/assets/42533446/eecc08c0-5b50-4909-b830-bb87baa840a7)

I tried the same in a vibrant orange. The proportions feel a little bit funny, but it feels good having the Haxophone a bit further away from the body. Next thing to try is sizing it more like a soprano, with the alto neck fitting of course.

**Reply by jcard0na** [reply]
> it feels good having the Haxophone a bit further away from the body

With FreeCAD it is trivial to make the neck longer, if that is what you want...
![long-neck](https://github.com/cardonabits/haxo-hw/assets/676181/7536c260-789d-4517-b521-232711e97e79)
